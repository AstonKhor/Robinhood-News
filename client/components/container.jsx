import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

import { NewsArticle } from './index.js'

const Box = styled.div`
  border-radius: 4px;
  margin: 0 -30px;
  padding: 30px;
  position: relative;
  transition: background 150ms;
`

const Container = (props) => {
  return(
    <Box>
      <NewsArticle />
    </Box>
  )
}

export default Container
