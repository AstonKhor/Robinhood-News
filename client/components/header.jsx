import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import moment from 'moment';

const Head = styled.div`
  margin-top: -4px;
  display: block;
  box-sizing: inherit;
`

const Source = styled.span`
color: ${props => {
  const now = moment();
  const totalMinutes = (now.hours() * 60) + now.minutes();
  let background = '';
  if (totalMinutes < 360 || totalMinutes >= 900) {
    return '#fff;'
  } else {
    return 'rgb(23, 23, 24);'
  }
  }
};
`

const Date = styled.span`
  color: #8c8c8e;
  margin-left: 6px;
  font-size: 13px;
  font-weight: 400px;
  letter-spacing: 0.25px;
  line-height: 19px;
`
const Header = (props) => {
  let { source, date } = props;
  return (
    <Head>
      <span className="bold">
        <Source>{source}</Source>
      </span>
      <Date>{moment(date).format('MMM D')}</Date>
    </Head>
  )
}

export default Header
