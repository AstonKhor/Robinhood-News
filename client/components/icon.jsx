import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const Eye = styled.svg`
  margin-right: 6px;
  overflow: visible;
`
const Icon = () => {
  return(
    <Eye
    width="14"
    height="11"
    viewBox="0 0 14 11"
    fill="8c8c8e"
    >
      <path fillRule="evenodd" d="M7 10.8889 C 3.13401 10.8889 0 7.04766 0 5.44444 C 9.81687e-17 3.84123 3.13401 0 7 0 C 10.866 0 14 3.84313 14 5.44444 C 14 7.04576 10.866 10.8889 7 10.8889 Z M 7 7.77778 C 8.28866 7.77778 9.33333 6.73311 9.33333 5.44444 C 9.33333 4.15578 8.28866 3.11111 7 3.11111 C 5.71134 3.11111 4.66667 4.15578 4.66667 5.44444 C 4.66667 6.73311 5.71134 7.77778 7 7.77778 Z" />
    </Eye>
  )
}

export default Icon
