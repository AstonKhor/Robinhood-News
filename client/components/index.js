import Container from './container';
import NewsArticle from './newsArticle';
import Header from './header';
import TextBody from './textBody';
import Icon from './icon';

export { Container, NewsArticle, Header, TextBody, Icon }
