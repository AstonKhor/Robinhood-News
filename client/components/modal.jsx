import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import { Thumbnail } from './index.js'

const ConstructedModal = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1051;
  opacity: 0;
  transition: opacity linear 0.25s;
  width: ${props => {
    switch (props.size) {
      case "lg":
        return "800";
      default:
        return "480";
    }
  }}px;
  margin: 40px auto;
  border-radius: 5px;
  &.fade-in {
    opacity: 1;
    transition: opacity linear .25s;
  }
  &.fade-out {
    opacity: 0;
    transition: opacity linear .25s;
  }
`

const ModalContent = styled.div`
  z-index: 1050;
  width: 100%;
  background-color: #fefefe;
  box-shadow: 0 3px 9px rgba(0, 0, 0, 0.5);
  padding: 15px;
`

const Header = styled.div`
  width: 100%;
  text-align: center;
  padding: 15px;
`

const Background = styled.div`
  background: rgba(0,0,0,0.5);
  position: fixed;
  z-index: 1040;
  display: block;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  outline: 0;
`

const Body = styled.div`
`

const Source = styled.div``
const modalRoot = document.getElementById("modal-root");
class Modal extends React.Component {
  constructor(props){
    super()
    let { source, body, title, thumbnail, views, date } = props.article
    this.state = {
      source: source,
      body: body,
      title: title,
      thumbnail: thumbnail,
      views: views,
      date: date,
      fadeType: null
    }
    this.background = React.createRef();
    this.handleClick = this.handleClick.bind(this)
    this.transitionEnd = this.transitionEnd.bind(this)
  }
  componentDidMount() {
    setTimeout(() => this.setState({ fadeType: "in" }), 0);
    let data = {idx: this.props.idx}
    fetch(`/news/addViews${window.location.search}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(data),
    })
      .then((response) => {
        return response.text()
      }).then( (response) => {
        console.log(response)
      })
  }

  componentDidUpdate(prevProps, prevState) {
    if (!this.props.isOpen && prevProps.isOpen) {
      this.setState({ fadeType: "out" });
    }
  }

  transitionEnd(e) {
    if (e.propertyName !== "opacity" || this.state.fadeType === "in") return;
    if (this.state.fadeType === "out") {
      this.props.onClose();
    }
  };

  handleClick(e){
    e.preventDefault();
    this.setState({ fadeType: "out" });
  }

  render() {
    let { source, body, title, thumbnail, views, date, fadeType } = this.state
    return ReactDOM.createPortal(
      <ConstructedModal
        id={this.props.id}
        className={`wrapper size-md fade-${fadeType}`}
        modalSize={""}
        onTransitionEnd={this.transitionEnd}
        >
        <ModalContent>
          <Header>
            {title}
          </Header>

          <Body>
            {body}
          </Body>
          <br></br>
          <Source>
            {source}
          </Source>
          <Thumbnail image={thumbnail} />
        </ModalContent>
        <Background
          onClick={this.handleClick}
          ref={this.background}
        />
        </ConstructedModal>,
      modalRoot
    );
    }
}
export default Modal
