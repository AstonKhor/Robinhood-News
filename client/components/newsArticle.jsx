import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import { Header, TextBody, Icon, Thumbnail } from './index.js'

const Box = styled.div`
  border-radius: 4px;
  position: relative;
  transition: bacgkround 150ms;
  display: flex;
  min-height: 134px;
`

const Text = styled.div`
  padding-right: 0;
  flex-direction: column;
  flex-grow: 1;
  display: flex;
`
const ViewsBox = styled.div`
  margin-top: auto;
  line-height: 19px;
  color: #8c8c8e;
  fill: #8c8c8e;
`
const NewsArticle = (props) => {
  let { source, body, title, thumbnail, views, date } = !!props.title ? props : ""

  return(
    <Box>
      <Text>
        <Header source={source} date={date} />
        <TextBody title={title} body={body} />
        <ViewsBox>
          <Icon />{views}
        </ViewsBox>
      </Text>

      <Thumbnail image={thumbnail} />
    </Box>
  )
}

export default NewsArticle
