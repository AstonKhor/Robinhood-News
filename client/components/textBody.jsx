import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';


const Box = styled.div`
  margin: 4px 0;
  max-height: 10.5ex;
  max-width: 420px;
  overflow: hidden;
  box-sizing: inherit;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
`
const Spacer = styled.div`
  margin-bottom: 2px;
`

const Title = styled.h3`
  margin-bottom: 2px;
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 22px;
  margin: 0;
  color: #fff;
`
const Body = styled.div`
  color: #8c8c8e;
  display: block;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin: 4px 0;
  max-height: 10.5ex;
  max-width: 420px;
  overflow: hidden;
`

const TextBody = (props) => {
  let { body, title } = props
  return (
    <Box>
      <Title>
        <Spacer>{title}</Spacer>
      </Title>
      <Body>
        {body}
      </Body>
    </Box>
  )
}

export default TextBody
