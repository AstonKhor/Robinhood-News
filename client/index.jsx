import React from 'react';
import ReactDOM from 'react-dom';
import { Container, Modal } from './components';
import styled from 'styled-components'
import GlobalStyle from './globalStyle';

const Title = styled.div`
  color: #fff;
  font-size: 26px;
  font-weight: 500;
  letter-spacing: -0.14px;
  line-height: 30px;
  border-bottom: 1px solid #0e0d0d;
  padding-bottom: 16px;
`;

const Body = styled.div`
  display: flex;
`
const Inner = styled.div`
  width: 750px;
  margin: 0 auto;
`

class App extends React.Component {
  constructor() {
    super()

    this.state = {
      data: [],
      showModal: false,
      modalIdx: null
    }

    this.toggleModal = this.toggleModal.bind(this)
  }

  toggleModal(idx) {
    this.setState({
      showModal: !this.state.showModal,
      modalIdx: idx
    })
  }

  componentDidMount() {
    fetch(`/news/getData${window.location.search}`)
      .then((response) => {
        return response.json()
      })
      .then((body) => {
        console.log(body)
        this.setState({
          data: body
        })
      })
      .catch((err) => {
        console.log(err)
      })
  }

  render() {
    return (
      <Body>
        <GlobalStyle />
        <Inner>
        <Title>News</Title>
        <Container data={this.state.data} modalHandler={this.toggleModal} />
        </Inner>
        {this.state.showModal && (
          <Modal
            id="modal"
            isOpen={this.state.showModal}
            onClose={this.toggleModal}
            article={this.state.data[this.state.modalIdx]}
            idx={this.state.modalIdx}
          />
        )}
      </Body>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('News'))
