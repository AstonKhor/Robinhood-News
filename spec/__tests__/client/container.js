import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// still need to make this folder when building the frontend.
import { NewsArticle , Container, Header, Thumbnail } from '../../../client/components';

describe('Container', () => {
  it('should render with no problems', () => {
    const component = renderer.create(
      <Container
        data={[
          {
            source: "google",
            body: "lorem ipsum",
            title: "Twitter",
            thumbnail: "image.url",
            views: 932,
            date: Date.now
          }
        ]
      }  />
    );
    expect(component).toMatchSnapshot();
  })

  it('should contain multiple news objects for the company', () => {
    let articles = [
      {
        title: "article1"
      },
      {
        title: "article2"
      }
    ]
    const component = mount(<Container data={articles} />)
    expect(component.find('NewsArticle').length).toBe(2)
  })

  it('renders other static components', () => {
    const component = mount(<Container />)

    expect(component.find(NewsArticle)).not.toBeNull()
  })
})
