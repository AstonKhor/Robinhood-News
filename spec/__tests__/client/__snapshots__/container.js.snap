// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with no problems 1`] = `null`;
